packages/connect/src/protocol/invoke-implementation.ts

```diff
@@ -149,8 +149,16 @@ export function transformInvokeImplementation<
         const input2 = await inputIt.next();
         if (input2.done !== true) {
           throw new ConnectError(
-            "protocol error: received extra input message for unary method",
+            `protocol error: received extra input message ${spec.method.I.typeName} for unary method ${spec.method.name}`,
             Code.Unimplemented,
+            {
+              methodName: spec.method.name,
+              serviceTypeName: spec.service.typeName,
+              inputTypeName: spec.method.I.typeName,
+              outputTypeName: spec.method.O.typeName,
+              url: context.url,
+              requestMethod: context.requestMethod,
+            }
```

Comment:
Metadata is used for errors returned as part of the protocol, fully qualified method name should be sufficient to pin point the rpc

Reply:
If I understand correctly, you are suggesting I remove this as adding metadata would interfere with other parts of the stack?

I can do that. I agree that the rpc method is the main piece of information missing.